components/brave_wallet/browser/rlp_decode.h

```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_RLP_DECODE_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_RLP_DECODE_H_
+
+#include <string>
+
+#include "base/values.h"
+
+namespace brave_wallet {
+
+// Recursive Length Prefix (RLP) decoding of arbitrarily nested arrays of data
+bool RLPDecode(const std::string& s, base::Value* output);
```

Comment:
I can see 0x is stripped in test cases so it is better to specify the input format in comment.

Reply:
Fixed in 51302353876330564318ed3f22ff8bf7bc1f99e9